mlflow/gateway/providers/gemini.py

```diff
@@ -1,12 +1,242 @@
+                    ),
+                    finish_reason=finish_reason,
+                )
+            )
+
+        usage_metadata = resp.get("usageMetadata", {})
+        prompt_tokens = usage_metadata.get("promptTokenCount", None)
+        completion_tokens = usage_metadata.get("candidatesTokenCount", None)
+        total_tokens = usage_metadata.get("totalTokenCount", None)
+
+        return chat.ResponsePayload(
+            id=resp.get("promptFeedback", {}).get("promptId", f"gemini-chat-{int(time.time())}"),
+            created=int(time.time()),
+            object="chat.completion",
+            model=config.model.name,
+            choices=choices,
+            usage=chat.ChatUsage(
+                prompt_tokens=prompt_tokens,
+                completion_tokens=completion_tokens,
+                total_tokens=total_tokens,
+            ),
+        )
+
+    @classmethod
+    def completions_to_model(cls, payload, config):
+        # Documentation: https://ai.google.dev/api/generate-content
+        # Example payload for the completions API.
+        #
+        # {
+        #     "prompt": "What is the world record for flapjack consumption in a single sitting?",
+        #     "temperature": 0.1,
+        #     "topP": 1,
+        #     "stop": ["\n"],
+        #     "n": 1,
+        #     "max_tokens": 100,
+        #     "top_k": 40,
+        # }
+
+        chat_payload = {"messages": [{"role": "user", "content": payload.pop("prompt")}], **payload}
+        return cls.chat_to_model(chat_payload, config)
+
+    @classmethod
+    def model_to_completions(cls, resp, config):
+        # Documentation: https://ai.google.dev/api/generate-content
+        #
+        # Example Response:
+        # {
+        #   "candidates": [
+        #     {
+        #       "content": {
+        #         "parts": [
+        #           {
+        #             "text": "Blue is often seen as a calming and soothing color."
+        #           }
+        #         ]
+        #       },
+        #       "finishReason": "stop"
+        #     }
+        #   ],
+        #   "usageMetadata": {
+        #     "promptTokenCount": 10,
+        #     "candidatesTokenCount": 10,
+        #     "totalTokenCount": 20
+        #   }
+        # }
+        choices = []
+
+        for idx, candidate in enumerate(resp.get("candidates", [])):
+            text = ""
+            if "content" in candidate and candidate.get("content", {}).get("parts", {}):
+                text = candidate["content"]["parts"][0].get("text", "")
+
+            finish_reason = candidate.get("finishReason", "stop")
+            if finish_reason == "MAX_TOKENS":
+                finish_reason = "length"
+
+            choices.append(
+                completions.Choice(
+                    index=idx,
+                    text=text,
+                    finish_reason=finish_reason,
+                )
+            )
+
+        usage_metadata = resp.get("usageMetadata", {})
```

Comment:
nit: can we avoid defining the variable here and just directly use `usage_metadata.get("xxx", None)` in the return statement below?